src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -0,0 +1,107 @@
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ
+{
+    [Extension("RabbitMQ")]
+    internal class RabbitMQExtensionConfigProvider : IExtensionConfigProvider
+    {
+        private readonly IOptions<RabbitMQOptions> _options;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public RabbitMQExtensionConfigProvider(IOptions<RabbitMQOptions> options, ILoggerFactory loggerFactory)
+        {
+            _options = options;
+            _loggerFactory = loggerFactory;
+        }
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            var rule = context.AddBindingRule<RabbitMQAttribute>();
+            rule.AddValidator(this.ValidateBinding);
+            rule.BindToCollector<byte[]>((attr) =>
+            {
+                return new RabbitMQAsyncCollector(this.CreateContext(attr), _loggerFactory.CreateLogger<RabbitMQAsyncCollector>());
+            });
+            rule.AddConverter<string, byte[]>(msg => Encoding.UTF8.GetBytes(msg));
+            rule.AddOpenConverter<OpenType.Poco, byte[]>(typeof(PocoToStringConverter<>));
+        }
+
+        public void ValidateBinding(RabbitMQAttribute attribute, Type type)
+        {
+            string hostname = Utility.FirstOrDefault(attribute.Hostname, _options.Value.Hostname);
+            string queuename = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName);
+            string exchange = Utility.FirstOrDefault(attribute.Exchange, _options.Value.Exchange);
+            IBasicProperties properties = attribute.Properties;
+
+            if (string.IsNullOrEmpty(hostname))
+            {
+                throw new InvalidOperationException("RabbitMQ hostname is missing");
+            }
+
+            if (string.IsNullOrEmpty(queuename))
+            {
+                throw new InvalidOperationException("RabbitMQ queuename is missing");
+            }
+
+            if (exchange == null)
+            {
+                exchange = string.Empty;
+            }
+        }
+
+        internal RabbitMQContext CreateContext(RabbitMQAttribute attribute)
+        {
+            string hostname = Utility.FirstOrDefault(attribute.Hostname, _options.Value.Hostname);
+            string queuename = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName);
+            string exchange = Utility.FirstOrDefault(attribute.Exchange, _options.Value.Exchange);

```

Comment:
change to:
`string exchange = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName) ?? string.Empty;`